### PR TITLE
fix: normalize old_protocol for E2

### PR DESCRIPTION
**Pull request by chemelli74**
## Summary by CodeRabbit

- **New Features**
  - Introduced a new `OldProtocol` enum with values `auto`, `true`, and `false`.
  - Enhanced the logic for handling old protocol settings.

- **Refactor**
  - Improved the `old_protocol` method, now `_normalize_old_protocol`, for better performance and reliability.
  - Updated the `set_customize` method to more robustly handle the `old_protocol` parameter.

### CODE REVIEW

File: midealocal/devices/e2/__init__.py [hunk 1]
```diff
@@ -74,12 +81,22 @@ def __init__(
...
         self._old_protocol = self._default_old_protocol
         self.set_customize(customize)
 
-    def old_protocol(self) -> bool:
-        return self.subtype <= 82 or self.subtype == 85 or self.subtype == 36353
+    def _normalize_old_protocol(self, value: Any) -> OldProtocol:
+        try:
+            if isinstance(value, str):
+                value = OldProtocol(value)
+            elif isinstance(value, int):
```
Comment (rokam) [inline midealocal/devices/e2/__init__.py:92]:
```suggestion
            elif isinstance(value, int) or isinstance(value, bool):

```

**Review by rokam — CHANGES_REQUESTED** [review]
We should consider it bool too

File: midealocal/devices/e2/__init__.py [hunk 2]
```diff
@@ -74,12 +81,22 @@ def __init__(
...
+                value = OldProtocol.true if value else OldProtocol.false
+            if not isinstance(value, OldProtocol):
```
Comment (rokam) [inline midealocal/devices/e2/__init__.py:94]: Better remove unneeded comparison
```suggestion
            else:
```

**Review by rokam — DISMISSED** [review]

**Comment by chemelli74** [comment]
> We should consider it bool too

Done, even if I think that once the user enter False in the dialog is will still be saved as string.

**Review by Necroneco — APPROVED** [review]

**Comment by Necroneco** [comment]
> Done, even if I think that once the user enter False in the dialog is will still be saved as string.

The user input is a complete JSON string, it will be parsed as expected.

**Review by rokam — APPROVED** [review]

**Pull request merged**